scripts/criu-ns

```diff
@@ -139,6 +155,23 @@ def wrap_restore():
         _mount_new_proc()
         run_criu(restore_args)
 
+    if restore_pidfile:
+        restored_pid = None
+        retry = 5
+
+        while not restored_pid and retry:
+            with open('/proc/%d/task/%d/children' % (criu_pid, criu_pid)) as f:
+                line = f.readline().strip()
+                if len(line):
+                    restored_pid = line
+                    break
+            retry -= 1
+            time.sleep(1)
+
+        if restored_pid:
+            with open(restore_pidfile, 'w+') as f:
+                f.write(restored_pid)
```

Comment:
I believe that CRIU creates the pidfile in the path specified with `--work-dir`, or if not specified it uses the path from `--images-dir` path.

@Snorch Could we get this path from `/proc/<PID>/cwd`?
https://github.com/checkpoint-restore/criu/blob/166aac0fb2673336e3cfa2190fedface0443bc27/criu/crtools.c#L252

Reply:
I think the `pidfile` should be created in the path specified by `--images-dir`. However, changes introduced in this PR create the `pidfile` in `criu/test/others/criu-ns`. It should be in `criu/test/others/criu-ns/dumpdir`

Reply:
Relative paths handling in CRIU is really bad to my taste, normally the caller of some command expects relative paths given in arguments be relative to cwd.

In criu we actually can save cwd in fdstore and resolve relative path with openat relative to it. Maybe it would be more logical. Sounds like a good project for GSOC.

Currently predicting there criu will put some file is really hard, it can put it to workdir or images dir or / if we save it after setns(CLONE_NEWNS) or some other syscall which affects cwd....

I see in depopulate_roots_yard that we save and restore cwd around setns, but we don't restore root, meaning that instead absolute paths can break, e.g. If originally criu was called from chroot...

But yes it seems for me that currently pidfile is put as you say, will try to do same in criu-ns.